iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/provider/ConfigBasedIcebergCatalogWrapperProvider.java

```diff
@@ -75,7 +75,7 @@ public IcebergCatalogWrapper getIcebergTableOps(String catalogName) {
       LOG.warn(errorMsg);
       throw new RuntimeException(errorMsg);
     }
-    return new IcebergCatalogWrapper(icebergConfig);
+    return new IcebergCatalogWrapper(icebergConfig, true);
```

Comment:
Why do we need to have a boolean to control whether to use credential vending or not, is there any better way?

Also, I saw there're lot of hard-code value in this PR, if it is enabled by default, maybe we can figure out a better way other than hard-coded boolean.

Reply:
removed  credential vending logic from  `IcebergCatalogWrapper`, do credential vending after related operation is finished outside.